src/index.ts

```diff
@@ -229,17 +231,17 @@ function encodeType(
 function findTypeDependencies(
   primaryType: string,
   types: Record<string, MessageTypeProperty[]>,
-  results: string[] = [],
-): string[] {
+  results: Set<string> = new Set(),
+): Set<string> {
   [primaryType] = primaryType.match(/^\w*/u);
-  if (results.includes(primaryType) || types[primaryType] === undefined) {
+  if (results.has(primaryType) || types[primaryType] === undefined) {
     return results;
   }
-  results.push(primaryType);
+
+  results.add(primaryType);
+
   for (const field of types[primaryType]) {
-    for (const dep of findTypeDependencies(field.type, types, results)) {
-      !results.includes(dep) && results.push(dep);
-    }
```

Comment:
`results` is the same object throughout the entire recursion!

Reply:
Haha, oh, good find.